Mathlib/MeasureTheory/Measure/Trim.lean

```diff
@@ -0,0 +1,136 @@
+/-
+Copyright (c) 2023 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne
+-/
+import Mathlib.MeasureTheory.Measure.MeasureSpace
+
+/-!
+# Restriction of a measure to a sub-σ-algebra
+
+It is common to see a measure `μ` on a measurable space structure `m0` as being also a measure on
+any `m ≤ m0`. Since measures in mathlib have to be trimmed to the measurable space, `μ` itself
+cannot be a measure on `m`, hence the definition of `μ.trim hm`.
+
+## Main definitions
+
+* `MeasureTheory.Measure.trim`: restriction of a measure to a sub-sigma algebra.
+
+-/
+
+open scoped ENNReal
+
+namespace MeasureTheory
+
+variable {α : Type*}
+
+/-- Restriction of a measure to a sub-σ-algebra.
+It is common to see a measure `μ` on a measurable space structure `m0` as being also a measure on
+any `m ≤ m0`. Since measures in mathlib have to be trimmed to the measurable space, `μ` itself
+cannot be a measure on `m`, hence the definition of `μ.trim hm`.
+
+This notion is related to `OuterMeasure.trim`, see the lemma
+`toOuterMeasure_trim_eq_trim_toOuterMeasure`. -/
```

Comment:
I'm not sure it's worth repeating this in the module dostring, since the two appear right after each other in the docs anyway.